src/transformState.js

```diff
@@ -5,7 +5,28 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const version of actions) {
+    switch (version.type) {
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      case 'addProperties':
+        Object.assign(state, version.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const removable of version.keysToRemove) {
+          delete state[removable];
+        }
+        break;
+
+      default:
+        break;
```

Comment:
maybe it is better to use `throw` instead of break?

Reply:
you can implement it